astropy/units/core.py

```diff
@@ -2098,14 +2100,28 @@ def __call__(
                 # or `s` was a composite unit (ValueError).
                 pass
 
+            warning_actions = {"silent": "ignore", "warn": "default", "raise": "error"}
+            if parse_strict not in warning_actions:
```

Comment:
How about adding this inside the `try/except`:
```
except KeyError as err:
    if parse_strict not in warning_actions:
        raise ValueError(...) from None
    else:
        raise
...
```
Saves a bit of time.

Also, is python smart enough not to re-evaluate the dict each time? Or is it better to define it outside?

Reply:
I believe it re-evaluates unless the function is compiled in some way, e.g. numba / jax.

Reply:
I had placed the `if parse_strict not in warning_actions:` check outside the `try-except` code because I wanted to exclude the possibility of accidentally catching a wrong `KeyError`, but putting the check inside the `except KeyError` block achieves that too.